source/Group7_GP_Agent/agent.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file agent.hpp
+ * @author Aman Dhruva Thamminana (thammina@msu.edu)
+ * @brief this will be our agent class can be created
+ * @version 0.0.1 [internal]
+ * @date 2023-09-25
+ *
+ * @copyright Copyright (c) 2023
+ *
+ */
+
+#pragma once
+
+#include <cassert>
+#include <iostream>
+#include <map>
+#include <string>
+
+#include <random>
+
+#include "../core/AgentBase.hpp"
+#include "./GPAgentSensors.hpp"
+
+/**
+ * @brief yeeeeeeeehaaaaaaaaa 🤠
```

Comment:
We probably shouldn't clutter the code with unnecessary and uninformative comments.  If you want to say something like "Main agent class for Group 7" and then the bit of flavor, that's probably okay, but flavor by itself (without any helpful explanation) is just distracting.